File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/homework/HoweWorkQSTReponseAdminService.java
package ma.zs.zyn.service.facade.admin.homework;

import java.util.List;
import ma.zs.zyn.bean.core.homework.HoweWorkQSTReponse;
import ma.zs.zyn.dao.criteria.core.homework.HoweWorkQSTReponseCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface HoweWorkQSTReponseAdminService extends  IService<HoweWorkQSTReponse,HoweWorkQSTReponseCriteria>  {

    List<HoweWorkQSTReponse> findByHomeWorkQuestionId(Long id);
    int deleteByHomeWorkQuestionId(Long id);
    long countByHomeWorkQuestionRef(String ref);




}

File: frontend-bqckup/src/app/shared/service/admin/homework/HoweWorkQSTReponseAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {HoweWorkQSTReponseDto} from '../../../model/homework/HoweWorkQSTReponse.model';
import {HoweWorkQSTReponseCriteria} from '../../../criteria/homework/HoweWorkQSTReponseCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class HoweWorkQSTReponseAdminService extends AbstractService<HoweWorkQSTReponseDto, HoweWorkQSTReponseCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/howeWorkQSTReponse/';
    }

    public constrcutDto(): HoweWorkQSTReponseDto {
        return new HoweWorkQSTReponseDto();
    }

    public constrcutCriteria(): HoweWorkQSTReponseCriteria {
        return new HoweWorkQSTReponseCriteria();
    }
}
